algebird-core/src/main/scala/com/twitter/algebird/BloomFilter.scala

```diff
@@ -225,6 +236,12 @@ case class BFInstance(hashes: BFHash, bits: BitSet, width: Int) extends BF {
       width)
   }
 
+  def checkAndAdd(item: String): (BF, ApproximateBoolean) = {
+    val bitsToActivate = BitSet(hashes(item): _*)
+    val contained = this.contains(item)
+    (BFInstance(hashes, bits ++ bitsToActivate, width), contained)
+  }
+
   def bitSetContains(bs: BitSet, il: Int*): Boolean = {
     il.map{ i: Int => bs.contains(i) }.reduce{ _ && _ }
```

Comment:
this could be optimized by stopping at the first false:
```scala
val ilit = il.iterator
var contains = true
while(ilit.hasNext && contains) {
  contains = contains && bs.contains(ilit.next)
}
contains
```
it is mutable, but it might speed things up measurably.